internal/services/recoveryservices/backup_protected_vm_resource.go

```diff
@@ -105,8 +106,19 @@ func resourceRecoveryServicesBackupProtectedVMCreateUpdate(d *pluginsdk.Resource
 		}
 
 		if !response.WasNotFound(existing.HttpResponse) {
-			return tf.ImportAsExistsError("azurerm_backup_protected_vm", id.ID())
+			isSoftDeleted := false
+			if meta.(*clients.Client).Features.RecoveryServicesVault.RecoverSoftDeletedBackupProtectedVM {
+				isSoftDeleted, err = resourceRecoveryServicesVaultBackupProtectedVMRecoverSoftDeleted(ctx, client, opClient, id, existing.Model)
+				if err != nil {
+					return fmt.Errorf("recovering soft deleted %s: %+v", id, err)
+				}
+			}
+
+			if !isSoftDeleted {
+				return tf.ImportAsExistsError("azurerm_backup_protected_vm", id.ID())
```

Comment:
For soft deleted resource that exists, we shall error out when users haven't opt in the soft delete feature, and ensure we provide a detailed error message as: https://github.com/hashicorp/terraform-provider-azurerm/blob/2d48f47beb482c40e30b12a1f77e95002e0b6cbe/internal/services/keyvault/key_vault_resource.go#L276-L284